plugins/uefi/efi/fwup-efi.c

```diff
@@ -17,16 +17,28 @@ fwup_delete_variable(CHAR16 *name, EFI_GUID *guid)
 {
 	EFI_STATUS rc;
 	UINT32 attrs = 0;
+	UINTN size = 0;
+	_cleanup_free VOID *buf = NULL;
 
 	/* get the attrs so we can delete it */
-	rc = uefi_call_wrapper(RT->GetVariable, 5, name, guid, &attrs, NULL, NULL);
+	rc = uefi_call_wrapper(RT->GetVariable, 5, name, guid, &attrs, &size, NULL);
 	if (EFI_ERROR(rc)) {
-		if (rc == EFI_NOT_FOUND) {
+		if (rc == EFI_BUFFER_TOO_SMALL) {
+			buf = fwup_malloc(size);
+			if (buf == NULL)
```

Comment:
why should we allocate the buffer for buf when we don't actually use it?

Reply:
When datasize is 0, will get EFI_BUFFER_TOO_SMALL and UEFI firemware will return the  needed buffer with datasize, and call GetVariable again to get attribute.

Reply:
UEFI really is terrible. Thanks for the sanity check.